### PR TITLE
segment-line: validate intersection with parent

### LINKED ISSUE #119: "segment-line" produces "GeometryCollection" (which has no coords)
I receive the following error when running `segment-line`:
```
Traceback (most recent call last):
  File "/home/kmw/OCR-D/env/bin/ocrd-tesserocr-segment-line", line 8, in <module>
    sys.exit(ocrd_tesserocr_segment_line())
  File "/home/kmw/OCR-D/env/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/kmw/OCR-D/env/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/kmw/OCR-D/env/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/kmw/OCR-D/env/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/kmw/OCR-D/env/lib/python3.7/site-packages/ocrd_tesserocr/cli.py", line 26, in ocrd_tesserocr_segment_line
    return ocrd_cli_wrap_processor(TesserocrSegmentLine, *args, **kwargs)
  File "/home/kmw/OCR-D/env/lib/python3.7/site-packages/ocrd/decorators.py", line 54, in ocrd_cli_wrap_processor
    run_processor(processorClass, ocrd_tool, mets, workspace=workspace, **kwargs)
  File "/home/kmw/OCR-D/env/lib/python3.7/site-packages/ocrd/processor/base.py", line 57, in run_processor
    processor.process()
  File "/home/kmw/OCR-D/env/lib/python3.7/site-packages/ocrd_tesserocr/segment_line.py", line 115, in process
    line_polygon = line_poly.exterior.coords
AttributeError: 'GeometryCollection' object has no attribute 'exterior'
```
on the image
FILE_0024_ORIGINAL

### PR DESCRIPTION
**Pull request by bertsky**
Fix #119

### CODE REVIEW

File: ocrd_tesserocr/segment_line.py [hunk 1]
```diff
@@ -111,7 +111,19 @@ def process(self):
...
                             # this could happen due to rotation
-                            line_poly = line_poly.intersection(region_poly).convex_hull
+                            interline = line_poly.intersection(region_poly)
+                            if interline.is_empty:
+                                continue # ignore this line
+                            if hasattr(interline, 'geoms'):
+                                # is (heterogeneous) GeometryCollection
+                                area = 0
+                                for geom in interline.geoms:
+                                    if geom.area > area:
```
Comment (wrznr) [inline ocrd_tesserocr/segment_line.py]: This selects the largest geom in the collection, right? This is not a union, right?

Comment (bertsky) [reply ocrd_tesserocr/segment_line.py]: Exactly. Shapely returns a GeometryCollection when the result of the intersection is discontiguous (which we cannot represent in PAGE-XML coordinates). Thus, a union would only yield inconsistency w.r.t. the parent region again. Sometimes, the result can even be _heterogeneous_, i.e. line and point, polygon and line etc.

I thought looking at the area would be the safest/simplest way to handle all cases. But I have no good examples either way.

**Comment by wrznr** [comment]
@bertsky Can we merge this?

**Comment by bertsky** [comment]
> Can we merge this?

If you have tested on your dataset to satisfaction, then yes. (As I said, I was not able to come up with enough tests myself.)

**Comment by bertsky** [comment]
> > Can we merge this?
> 
> If you have tested on your dataset to satisfaction, then yes. (As I said, I was not able to come up with enough tests myself.)

@wrznr have you tested this on your real-life dataset yet?

**Pull request merged**